### PR TITLE
#117: Add node / edge attribute getter & setter to CellComplex

### PR DESCRIPTION
**Pull request by josefhoppe**

### CODE REVIEW

File: toponetx/classes/cell_complex.py [hunk 1]
```diff
@@ -868,6 +868,114 @@ def get_filtration(self, name: str) -> Dict[Hashable | tuple, Any]:
                 d.update(i)
         return d
 
+    def set_node_attributes(
+        self,
+        values: Union[dict[Hashable, dict], dict[Hashable, Any]],
+        name: Optional[str],
+    ):
```
Comment (ffl096) [inline toponetx/classes/cell_complex.py:877]:
```suggestion
    ) -> None:
```

File: toponetx/classes/cell_complex.py [hunk 2]
```diff
@@ -870,6 +870,114 @@ def get_filtration(self, name: str) -> dict[Union[Hashable, tuple], Any]:
...
+        Parameters
+        ----------
+        values :  dict
+            either contains node -> value (if `name` is specified)
+            or nested dict with node -> (attribute -> value) (if `name == None`)
+        name : str, optional
+
+        Returns
+        -------
+        None.
```
Comment (ffl096) [inline toponetx/classes/cell_complex.py:889]: Functions that don't return anything don't have a `Returns` section at all as per NumPy's documentation style.

File: toponetx/classes/cell_complex.py [hunk 3]
```diff
@@ -870,6 +870,114 @@ def get_filtration(self, name: str) -> dict[Union[Hashable, tuple], Any]:
...
+        >>> CX.get_node_attributes('color')
+        {1: 'red', 2: 'blue'}
+        """
+        return self.get_cell_attributes(rank=0, name=name)
+
+    def set_edge_attributes(
+        self,
+        values: Union[dict[tuple, dict], dict[tuple, Any]],
+        name: Optional[str],
+    ):
```
Comment (ffl096) [inline toponetx/classes/cell_complex.py:931]:
```suggestion
    ) -> None:
```

File: toponetx/classes/cell_complex.py [hunk 4]
```diff
@@ -870,6 +870,114 @@ def get_filtration(self, name: str) -> dict[Union[Hashable, tuple], Any]:
...
+        Parameters
+        ----------
+        values :  dict
+            either contains (node1, node2) -> value (if `name` is specified)
+            or nested dict with (node1, node2) -> (attribute -> value) (if `name == None`)
+        name : str, optional
+
+        Returns
+        -------
+        None.
```
Comment (ffl096) [inline toponetx/classes/cell_complex.py:943]: See above.

File: toponetx/classes/cell_complex.py [hunk 5]
```diff
@@ -870,6 +870,114 @@ def get_filtration(self, name: str) -> dict[Union[Hashable, tuple], Any]:
...
+        """Get node attributes.
+
+        Parameters
+        ----------
+        name : str
+
+        Returns
+        -------
+        attr :  dict of node -> value of attribute `name`
+            nodes without the given attribute are not in the dictionary.
```
Comment (ffl096) [inline toponetx/classes/cell_complex.py:913]: The documentation of the returned value is not as mandated by NumPy's documentation style.

1. The returned value doesn't have a name, so remove the `attr : ` from the first line.
2. The first line only describes the type, anything else is part of the comment starting in the second line. That the mapping is from nodes to value is not part of the type.

File: toponetx/classes/cell_complex.py [hunk 6]
```diff
@@ -870,6 +870,114 @@ def get_filtration(self, name: str) -> dict[Union[Hashable, tuple], Any]:
...
+        """Get edge attributes.
+
+        Parameters
+        ----------
+        name : str
+
+        Returns
+        -------
+        attr :  dict of edge (as tuple (node1, node2)) -> value of attribute `name`
+            edges without the given attribute are not in the dictionary.
```
Comment (ffl096) [inline toponetx/classes/cell_complex.py:967]: See above.

**Review by ffl096 — CHANGES_REQUESTED** [review]

File: toponetx/classes/cell_complex.py (hunk 5 above)
Comment (josefhoppe) [reply toponetx/classes/cell_complex.py:906]: Ah, it is like this everywhere. I also fixed other occasions I came accross

**Review by ffl096 — APPROVED** [review]

**Review by ninamiolane — APPROVED** [review]
LGTM!

**Pull request merged**